include/secp256k1_silentpayments.h

```diff
@@ -126,6 +127,50 @@ SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_silentpayments_create_t
     const unsigned char *input_hash
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4);
 
+/** Create Silent Payment shared secret.
+ *
+ * Given a public component Pub, a private component sec and an input_hash,
+ * calculate the corresponding shared secret using ECDH:
+ *
+ * shared_secret = (sec * input_hash) * Pub
+ *
+ * What the components should be set to depends on the role of the caller.
+ * For the sender side, the public component is set the recipient's scan public key
+ * B_scan, and the private component is set to the input's private keys sum:
+ *
+ * shared_secret = (a_sum * input_hash) * B_scan   [Sender]
+ *
+ * For the receiver side, the public component is set to the input's public keys sum,
+ * and the private component is set to the receiver's scan private key:
+ *
+ * shared_secret = (b_scan * input_hash) * A_sum   [Receiver, Full node scenario]
+ *
+ * In the "light client" scenario for receivers, the public component is already
+ * tweaked with the input hash: A_tweaked = input_hash * A_sum
+ * In this case, the input_hash parameter should be set to NULL, to signal that
+ * no further tweaking should be done before the ECDH:
+ *
+ * shared_secret = b_scan * A_tweaked   [Receiver, Light client scenario]
+ *
+ * The resulting shared secret is needed as input for creating silent payments
+ * outputs belonging to the same receiver scan public key.
+ *
+ *  Returns: 1 if shared secret creation was successful. 0 if an error occured.
+ *  Args:                  ctx: pointer to a context object
+ *  Out:       shared_secret33: pointer to the resulting 33-byte shared secret
+ *  In:       public_component: pointer to the public component
+ *           private_component: pointer to 32-byte private component
+ *                  input_hash: pointer to 32-byte input hash (can be NULL if the
+ *                              public component is already tweaked with the input hash)
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_silentpayments_create_shared_secret(
```

Comment:
in "silentpayments: add shared secret creation routine (a*B == A*b)" (https://github.com/bitcoin-core/secp256k1/pull/1471/commits/b0e37968b09b830b422bafb60c0e870d611ef6ed):

Building off the last couple suggestions, this function is now exclusively for light clients and there is no ambiguity about the arguments:

* `public_component` is always `A_tweaked`
* `private_component` is always `b_scan`

we can also drop the `input_hash` argument and change `public_component` to `unsigned char *A_tweaked`